app/models/changeset/pull_request.rb

```diff
@@ -3,11 +3,14 @@ class Changeset::PullRequest
   RISKS_SECTION = /#+\s+Risks.*\n/i
 
   # Matches URLs to JIRA issues.
-  JIRA_ISSUE = %r[https://\w+\.atlassian\.net/browse/[\w-]+]
+  JIRA_ISSUE_URL = %r[https?:\/\/[\da-z\.\-]+\.[a-z\.]{2,6}\/browse\/[A-Z][A-Z\d]+-\d+]
 
   # Matches "VOICE-1234" or "[VOICE-1234]"
-  JIRA_CODE = %r[(\[)*([a-zA-Z]+)-((\d)+)(\])*]
+  JIRA_CODE_TITLE = %r[(\[)*([a-zA-Z]+)-((\d)+)(\])*]
 
+  # Matches "VOICE-1234" only
+  PUNCT = "\\s|\\p{Punct}|~|="
+  JIRA_CODE = %r[(?<=#{PUNCT}|^)([A-Z][A-Z\d]+-\d+)(?=#{PUNCT}|$)]
```

Comment:
I think the comment is still accurate because the punctuations are look behind and look ahead only. I have moved out the code regex to a common one and added some examples of valid codes.